docs/dpdk/README.md

```diff
@@ -0,0 +1,74 @@
+# Running DPDK applications in Kubernetes
+
+## Pre-requisites
+
+### Hugepages
+DPDK applications require Hugepages memory. Please refer to the [Hugepages section](http://doc.dpdk.org/guides/linux_gsg/sys_reqs.html#use-of-hugepages-in-the-linux-environment) in DPDK getting started guide on hugespages in DPDK.
+
+Kubernetes nodes can only advertise a single size pre-allocated hugepages. Which means even though one can have both 2M and 1G hugepages in a system, Kubernetes will only recognize the default hugepages as schedulable resources. Workload can request for hugepages using resource requests and limits specifying `hugepage-2Mi` or `hugepage-1Gi` resource references.
+
+> One important thing to note here is that when requesting hugepage resources, either memory or CPU resource requests need to be specified.
+
+For more information on hugepage support in Kubernetes please see [here](https://kubernetes.io/docs/tasks/manage-hugepages/scheduling-hugepages/).
+
+
+### VF drivers
+DPDK applications require devices to be attached with supported dpdk backend driver.
+For Intel® x700 series NICs `igb_uio` or `vfio-pci` is required.
+
+### Privileges 
+Certain privileges are required for dpdk application to function properly in Kubernetes Pod. The level of privileges depend on the application and the host device driver attached.
+
+For example, devices with `igb_uio` driver requires a Pod to run with full privilege. With `vfio-pci` an application can run in a non-privilege Pod with only **IPC_LOCK** capability added.
+
+
+# Example deployment
+This directory includes sample deployment yaml files showing how to deploy a dpdk application in Kubernetes with in non-privileg Pod with SR-IOV VF attached to vfio-pci driver.
+
+## Prepare host for vfio support and hugepages memory
+
+1. On CentOS 7, edit `/etc/default/grub` file and add the following kernel boot parameters to enable iommu and create 8GB of 2M size hugepages.
+
+```
+GRUB_CMDLINE_LINUX="crashkernel=auto nomodeset rhgb quiet iommu=pt intel_iommu=on default_hugepagesz=2M hugepagesz=2M hugepages=4096 pci=realloc,assign-busses"
```

Comment:
nit: For 2M hugepages, it's possible to create hugepages without modifying cmdline and rebooting, might be good to use 1G hugepage size as example.

For example:

- /etc/default/grub 

`GRUB_CMDLINE_LINUX="crashkernel=auto rhgb quiet intel_iommu=on iommu=pt default_hugepagesz=1GB hugepagesz=1G hugepages=16"`

- kernel cmdline after rebooting

`BOOT_IMAGE=/vmlinuz-3.10.0-957.el7.x86_64 crashkernel=auto rhgb quiet intel_iommu=on iommu=pt default_hugepagesz=1GB hugepagesz=1G hugepages=16`

Reply:
> Is `pci=realloc,assign-busses` needed?

It may not be needed for all systems. This is to circumvent the "not enough MMIO resources" error on some systems.  Configurations here are from a working system(with various t/s done in the process for different issues)